operator/src/main/java/oracle/kubernetes/operator/helpers/JobHelper.java

```diff
@@ -392,50 +417,60 @@ public NextAction apply(Packet packet) {
+              info.getNamespace(), info.getDomain().getDomainUid(),
+              new ReplaceOrCreateStep(next));
     }
 
-    void logJobDeleted(String domainUid, String namespace, String jobName, Packet packet) {
-      V1Job domainIntrospectorJob =
-          (V1Job) packet.remove(ProcessingConstants.DOMAIN_INTROSPECTOR_JOB);
+    private class ReplaceOrCreateStep extends DefaultResponseStep {
 
-      packet.remove(ProcessingConstants.INTROSPECTOR_JOB_FAILURE_LOGGED);
-      if (domainIntrospectorJob != null
-          && !JobWatcher.isComplete(domainIntrospectorJob)) {
-        logIntrospectorFailure(packet, domainIntrospectorJob);
+      ReplaceOrCreateStep(Step next) {
+        super(next);
       }
-      packet.remove(ProcessingConstants.JOB_POD_NAME);
 
-      LOGGER.fine(getJobDeletedMessageKey(), domainUid, namespace, jobName);
-    }
+      @Override
+      public NextAction onSuccess(Packet packet, CallResponse callResponse) {
+        DomainPresenceInfo info = packet.getSpi(DomainPresenceInfo.class);
+        String namespace = info.getNamespace();
+        V1Job job = (V1Job) callResponse.getResult();
+        if ((job != null) && (packet.get(ProcessingConstants.DOMAIN_INTROSPECTOR_JOB) == null)) {
+          packet.put(ProcessingConstants.DOMAIN_INTROSPECTOR_JOB, job);
+        }
 
-    private Step deleteJob(Packet packet, Step next) {
-      DomainPresenceInfo info = packet.getSpi(DomainPresenceInfo.class);
-      java.lang.String domainUid = info.getDomain().getDomainUid();
-      java.lang.String namespace = info.getNamespace();
-      String jobName = JobHelper.createJobName(domainUid);
-      logJobDeleted(domainUid, namespace, jobName, packet);
-      return new CallBuilder().withTimeoutSeconds(JOB_DELETE_TIMEOUT_SECONDS)
-            .deleteJobAsync(
-                  jobName,
-                  namespace,
-                  domainUid,
-                  new V1DeleteOptions().propagationPolicy("Foreground"),
-                  new DefaultResponseStep<>(next));
+        if (job != null) {
+          packet.putIfAbsent(START_TIME, Optional.ofNullable(job.getMetadata())
+                  .map(m -> m.getCreationTimestamp()).map(t -> t.toInstant().toEpochMilli()).orElse(0L));
+          return doNext(Step.chain(
+                  createProgressingStartedEventStep(info, INSPECTING_DOMAIN_PROGRESS_REASON, true, null),
+                  readDomainIntrospectorPodLogStep(null),
+                  deleteDomainIntrospectorJobStep(null),
+                  ConfigMapHelper.createIntrospectorConfigMapStep(null),
+                  ConfigMapHelper.readExistingIntrospectorConfigMap(namespace, info.getDomainUid()),
+                  new DomainProcessorImpl.IntrospectionRequestStep(info),
+                  createDomainIntrospectorJobStep(getNext())), packet);
+        } else {
+          packet.putIfAbsent(START_TIME, System.currentTimeMillis());
```

Comment:
START_TIME could be an instance of OffsetDateTime.now(). There isn't a good reason to convert to a Long.

Reply:
Done in fedcece .